### PR TITLE
[wordpress__notices/wordpress__data/wordpress__blocks] Change `Notice` to `WPNotice` and update to correct props

### PR DESCRIPTION
**Pull request by opr**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/WordPress/gutenberg/blob/%40wordpress/notices%403.5.0/packages/notices/src/store/selectors.js#L19-L43
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## wordpress__blocks ([<kbd>unpkg</kbd>](https://unpkg.com/browse/wordpress__blocks@latest/))
was missing the following properties:
1. store
1. store
## wordpress__notices ([<kbd>unpkg</kbd>](https://unpkg.com/browse/wordpress__notices@latest/))
was missing the following properties:
1. store
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b462114f478e8d294d0189d814e2250ba2f18138
</p>

File: types/wordpress__notices/index.d.ts [hunk 1]
```diff
@@ -13,13 +13,47 @@ declare module '@wordpress/data' {
 }
 
 export type Status = 'error' | 'info' | 'success' | 'warning';
+export type WPNoticeType = 'snackbar' | 'default';
 
-export interface Notice {
+export interface WPNotice {
```
[the remark this answers is not in the document]
Comment (opr) [reply types/wordpress__notices/index.d.ts]: Right! I've reverted it now, thanks.

**Comment by opr** [comment]
hey @jonsurrell what are some worthwhile tests for my changes here? 

I think the way to test my changes would be to destructure a `Notice` and see if all the props are there, seems like it's not a very useful test?

**Comment by sirreal** [comment]
I'm not too familiar with this, but you can try to make a `Notice` with the additional props you've set and make sure that it's valid.

For the changes to `isValidBlockContent`, you might add a case where we expect an error if we don't satisfy the extends constraint around here:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1613a440814df80c4af8bd6aa4a3f02dc30d5e0a/types/wordpress__blocks/wordpress__blocks-tests.tsx#L598-L602

I'm not sure what other types we'd expect to see there:

```
// $ExpectError
blocks.isValidBlockContent('my/foo', 'invalid', 'Foobar');
```

**Comment by opr** [comment]
Sorry I've been super busy, I'll get around to adding tests soon! 🤞🏻

File: types/wordpress__notices/index.d.ts [hunk 2]
```diff
@@ -13,12 +13,46 @@ declare module '@wordpress/data' {
...
     content: string;
+    /**
+     * Audibly announced message text used by assistive technologies.
+     */
+    spokenMessage: string;
+    /**
+     * Notice message as raw HTML. Intended to serve primarily for compatibility of server-rendered notices, and SHOULD
+     * NOT be used for notices. It is subject to removal without notice.
+     */
+    __unstableHTML: string;
```
Comment (peterblazejewicz) [inline types/wordpress__notices/index.d.ts:39]: this looks like private by convention. Is this expected to be available in TS?

File: types/wordpress__notices/index.d.ts [hunk 3]
```diff
@@ -13,12 +13,46 @@ declare module '@wordpress/data' {
 }
 
 export type Status = 'error' | 'info' | 'success' | 'warning';
+export type WPNoticeType = 'snackbar' | 'default';
```
Comment (peterblazejewicz) [inline types/wordpress__notices/index.d.ts:16]:
```suggestion
export type NoticeType = 'snackbar' | 'default';
```
as in other place, there is no need really for custom prefix. Is the 'WP' used elsewhere for types here?

File: types/wordpress__notices/index.d.ts [hunk 4]
```diff
@@ -62,7 +96,7 @@ export interface Options {
      * The type of notice.
      * @defaultValue `'default'`
      */
-    type: 'default' | 'snackbar';
+    type: WPNoticeType;
```
Comment (peterblazejewicz) [inline types/wordpress__notices/index.d.ts:99]:
```suggestion
    type: NoticeType;
```
(same reason as in other place)

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Please address feedback

File: types/wordpress__notices/index.d.ts (hunk 2 above)
Comment (opr) [reply types/wordpress__notices/index.d.ts:39]: Thank you for the feedback!

I understand the concern, but in the test here https://github.com/WordPress/gutenberg/blob/%40wordpress/notices%403.5.0/packages/notices/src/store/test/actions.js#L90 it is used. On this basis I am not sure if removing it from the type definition is the correct thing to do if it is in fact a property that can be accessed?

If it was removed from the type definition then that code would be considered incorrectly typed wouldn't it?

File: types/wordpress__notices/index.d.ts (hunk 3 above)
Comment (opr) [reply types/wordpress__notices/index.d.ts:16]: No, I was basing my changes on the names used in the JSDoc style types in Gutenberg:

https://github.com/WordPress/gutenberg/blob/%40wordpress/notices%403.5.0/packages/notices/src/store/selectors.js#L19-L43

I have reverted it now though if it's not useful.

**Comment by opr** [comment]
Note that I removed the test of manually creating a `Notice` in commit 
91b4ea3 - it is not possible to do because of https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type/issues/18 -> https://github.com/microsoft/TypeScript/issues/9879

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**